packages/timeline/src/browser/timeline-widget.tsx

```diff
@@ -0,0 +1,165 @@
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+/* eslint-disable no-null/no-null, @typescript-eslint/no-explicit-any */
+
+import { Message } from '@phosphor/messaging';
+import { inject, injectable, postConstruct } from 'inversify';
+import { DisposableCollection } from '@theia/core/lib/common/disposable';
+import {
+    ApplicationShell,
+    BaseWidget,
+    MessageLoop,
+    Panel,
+    PanelLayout,
+    Widget
+} from '@theia/core/lib/browser';
+import { TimelineTreeWidget } from './timeline-tree-widget';
+import { EditorManager, EditorWidget } from '@theia/editor/lib/browser';
+import { TimelineService } from './timeline-service';
+import { CommandRegistry } from '@theia/core/lib/common';
+import { TabBarToolbarRegistry } from '@theia/core/lib/browser/shell/tab-bar-toolbar';
+import { TimelineEmptyWidget } from './timeline-empty-widget';
+import { toArray } from '@phosphor/algorithm';
+import URI from '@theia/core/lib/common/uri';
+import { EditorPreviewWidget } from '@theia/editor-preview/lib/browser';
+
+@injectable()
+export class TimelineWidget extends BaseWidget {
+
+    protected panel: Panel;
+
+    static ID = 'timeline-view';
+
+    @inject(TimelineTreeWidget) protected readonly resourceWidget: TimelineTreeWidget;
+    @inject(TimelineService) protected readonly timelineService: TimelineService;
+    @inject(TabBarToolbarRegistry) protected readonly tabBarToolbar: TabBarToolbarRegistry;
+    @inject(CommandRegistry) protected readonly commandRegistry: CommandRegistry;
+    @inject(ApplicationShell) protected readonly applicationShell: ApplicationShell;
+    @inject(TimelineEmptyWidget) protected readonly timelineEmptyWidget: TimelineEmptyWidget;
+
+    constructor(@inject(EditorManager) protected readonly editorManager: EditorManager) {
+        super();
+        this.id = TimelineWidget.ID;
+        this.addClass('theia-timeline');
+    }
+
+    @postConstruct()
+    protected init(): void {
+        const layout = new PanelLayout();
+        this.layout = layout;
+        this.panel = new Panel({
+            layout: new PanelLayout({
+            })
+        });
+        this.panel.node.tabIndex = -1;
+        layout.addWidget(this.panel);
+        this.containerLayout.addWidget(this.resourceWidget);
+        this.containerLayout.addWidget(this.timelineEmptyWidget);
+
+        this.refresh();
+        this.timelineService.onDidChangeTimeline(event => {
```

Comment:
Done, updated the description. It is interesting that `TimelineTreeWidget` still duplicates:
![Screenshot from 2020-07-07 15-14-32](https://user-images.githubusercontent.com/7668752/86779988-e33e6080-c064-11ea-9d7c-91fe62f666fe.png)
I don't have any ideas why it happens because `scmTreeWidget` binded in the same way but it doesn't have this problem. Do you have any ideas?